File: 2000/2812.cpp
#include<iostream>
#include<stack>
#include<algorithm>
using namespace std;

int n, k;
string num;

void input() {
	cin >> n >> k >> num;
}

void solve() {
	input();
	string answer = "";
	stack<char> s;
	n -= k;
	for (int i = 0; i < num.length(); ++i) {
		if (s.empty()) {
			s.push(num[i]);
			continue;
		}
		while (s.top() < num[i] && k > 0) {
			s.pop();
			k--;
			if (s.empty()) break;
		}
		s.push(num[i]);
	}
	while (!s.empty()) {
		answer += s.top();
		s.pop();
	}
	reverse(answer.begin(), answer.end());
	cout << answer.substr(0, n);
}

int main() {
	cin.tie(0), cout.tie(0);
	ios_base::sync_with_stdio(0);
	solve();
	return 0;
}

File: 2000/2812.java
import java.io.*;
import java.util.*;

public class Main {

    static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    static StringBuilder sb = new StringBuilder();
	static StringTokenizer st;
    static int n, k;
    static String num;

    static void input() throws IOException {
		st = new StringTokenizer(br.readLine());
		n = Integer.parseInt(st.nextToken());
		k = Integer.parseInt(st.nextToken());
		num = br.readLine();
    }
    
    static void solution() {
    	Stack<Character> s = new Stack<>();
		s.add(num.charAt(0));
		int len = n - k;
    	for(int i = 1;i < num.length(); ++i) {
    		char cur = num.charAt(i);
    		while(true) {
    			if(k == 0 || s.isEmpty()) break;
    			if(s.peek() < cur) {
    				--k;
    				s.pop();
    			} else break;
    		}
    		s.add(num.charAt(i));
    	}
    	while(!s.isEmpty()) {
    		sb.append(s.pop());
    	}
    	System.out.println(sb.reverse().substring(0, len));
    }
    
    public static void main(String[] args) throws IOException {
   		input();
   		solution();
    }
}
